evalml/pipelines/components/transformers/column_selectors.py

```diff
@@ -122,3 +123,44 @@ def transform(self, X, y=None):
             pd.DataFrame: Transformed X.
         """
         return super().transform(X, y)
+
+
+class SelectByTypeorTag(ColumnSelector):
```

Comment:
Mega-nit: `SelectByTypeOrTag`?  The lowercase or was bugging me

Reply:
I think `SelectByType` would be slightly easier to read without losing much meaning. Thoughts?

Reply:
Since the docstring mentions either logical type or semantic tag is possible, I'm fine renaming it to `SelectByType`!